payment-request/dynamically-change-shipping-options-manual.https.html

```diff
@@ -0,0 +1,111 @@
+<!DOCTYPE html>
+<!-- Copyright © 2017 Chromium authors and World Wide Web Consortium, (Massachusetts Institute of Technology, ERCIM, Keio University, Beihang). -->
+<meta charset="utf-8">
+<title>Test for PaymentRequest shippingOption dynamic updating</title>
+<link rel="help" href="https://w3c.github.io/payment-request/#shippingoption-attribute">
+<link rel="help" href="https://w3c.github.io/payment-request/#onshippingoptionchange-attribute">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+setup({ explicit_done: true, explicit_timeout: true });
+const validMethod = Object.freeze({ supportedMethods: "basic-card" });
+const applePayMethod = {
+  supportedMethods: "https://apple.com/apple-pay",
+  data: {
+    version: 3,
+    merchantIdentifier: "merchant.com.example",
+    countryCode: "US",
+    merchantCapabilities: ["supports3DS"],
+    supportedNetworks: ["visa"],
+  },
+};
+const validMethods = Object.freeze([validMethod, applePayMethod]);
+const validAmount = Object.freeze({ currency: "USD", value: "5.00" });
+const validTotal = Object.freeze({
+  label: "label",
+  amount: validAmount,
+});
+const validDetails = Object.freeze({ total: validTotal });
+
+const validShippingOption1 = Object.freeze({
+  id: "default-method",
+  label: "Default shipping method",
+  amount: validAmount,
+  selected: true,
+});
+
+const validDynamicShippingOption = Object.freeze({
+  id: "dynamically-added-id",
+  label: "Dynamically added shipping option",
+  amount: validAmount,
+  selected: false,
+});
+
+const requestShipping = Object.freeze({
+  requestShipping: true,
+});
+
+function testShippingOptionChanged() {
+  promise_test(async t => {
+    const detailsWithShippingOptions = Object.assign({}, validDetails, {
+      shippingOptions: [validShippingOption1],
+    });
+    const request = new PaymentRequest(
+      validMethods,
+      detailsWithShippingOptions,
+      requestShipping
+    );
+    const shippingAddressChangeListener = new Promise(resolve => {
+      request.addEventListener("shippingaddresschange", (ev) => {
+        // resolve(request.shippingOption);
+        ev.updateWith(Object.assign({}, {
+          shippingOptions: [validShippingOption1, validDynamicShippingOption]
+        }))
+        resolve()
+      }, {once:true});
+    });
+    const handlerPromise = new Promise(resolve => {
+      request.onshippingoptionchange = () => {
+        resolve(request.shippingOption);
+      };
+    });
+    request.show().catch(err => err);
+
+    const results = await Promise.all([shippingAddressChangeListener, handlerPromise]);
+    assert_true(
+      results[1] === "dynamically-added-id",
+      "Expected dynamically-added-id as the shippingOption"
+    );
+    await request.abort();
+  });
+  done();
```

Comment:
```suggestion
```

It's probably best to add done() as:

```HTML 
<button onclick="done()">Done</button>
``` 

After the test `<li>`... in case we want to add more tests in the future. Also, it prevents `done()` being called before the test has actually finished. 

I know this is probably wrong in other tests... we will need to fix that in the future.

Reply:
added the button, and added some clarifying text in the instructions as well just to be sure.